benchmark/Streamly/Benchmark/Data/Unfold.hs

```diff
@@ -131,57 +410,122 @@ zipWithM size start =
 moduleName :: String
 moduleName = "Data.Unfold"
 
-o_1_space_serial_generation :: Int -> [Benchmark]
-o_1_space_serial_generation size =
-    [ bgroup "generation"
-        [ benchIO "fromListM" $ fromListM size
-        , benchIO "replicateM" $ replicateM size
-        , benchIO "repeatM" $ repeatM size
-        , benchIO "enumerateFromStepIntegral" $ enumerateFromStepIntegral size
-        , benchIO "enumerateFromToIntegral" $ enumerateFromToIntegral size
-        ]
+o_1_space_generation_input_ops :: Int -> [Benchmark]
+o_1_space_generation_input_ops size =
+    [ bgroup
+          "generation/input-ops"
+          [ benchIO "lmap" $ lmap size
+          , benchIO "lmapM" $ lmapM size
+          , benchIO "supply" $ supply size
+          , benchIO "supplyFirst" $ supplyFirst size
+          , benchIO "supplySecond" $ supplySecond size
+          , benchIO "discardFirst" $ discardFirst size
+          , benchIO "discardSecond" $ discardSecond size
+          , benchIO "swap" $ swap size
+          ]
     ]
 
-o_1_space_serial_transformation :: Int -> [Benchmark]
-o_1_space_serial_transformation size =
-    [ bgroup "transformation"
-        [ benchIO "take" $ take size
-        , benchIO "takeWhileM" $ takeWhileM size
-        , benchIO "filterAllOut" $ filterAllOut size
-        , benchIO "filterAllIn" $ filterAllIn size
-        , benchIO "filterSome" $ filterSome size
-        -- This will take nanoseconds, We need to fix the benchmark reporting to
-        -- have microseconds as the minimum unit before uncommenting this.
-        -- , benchIO "dropOne" $ dropOne size
-        , benchIO "dropAll" $ dropAll size
-        , benchIO "dropWhileMTrue" $ dropWhileMTrue size
-        , benchIO "dropWhileMFalse" $ dropWhileMFalse size
-        ]
+o_1_space_generation_resource :: Int -> [Benchmark]
+o_1_space_generation_resource size =
+    [ bgroup
+          "generation/resource"
+          [ benchIO "fromStream" $ fromStream size
+          , benchIO "fromStreamK" $ fromStreamK size
+          -- XXX INVESTIGATE, see the definition
+          -- , benchIO "fromStreamD" $ fromStreamD size
```

Comment:
I guess this is just going to remain quadratic, we can have this benchmark run for a smaller stream size e.g. "fromStreamD (n/100)" or something like that and remove the `XXX` comment.